### DISCUSSION TITLE
socketio for NON web-based process communication

### DISCUSSION THREAD

**Discussion by rozensl2** [miguelgrinberg/python-socketio#887] [Q&A]
Hi there, 

I need to create a channel of communication between an Adobe application plug-in (Javascript) and a Python module. The plug-in sends data to the Python module, which processes this information and sends it back. The plug-in does not support the spawning of child processes, so I cannot use the "child-process" module. 

At the moment, the only way I can think of for these two processes to communicate is via WebSocket, as the plug-in supports it. Would python-socketio be able to connect to a javascript process running the WebSocket protocol? Please note that both processes will be running on the same machine and there will be no Internet connectivity.

**Comment by miguelgrinberg** [comment]
This package does not support plain WebSocket communication. The Socket.IO protocol is much more complex than WebSocket, it only uses WebSocket as a transport. If you don't have Socket.IO support on the Adobe side, then this package will not work for you.

**Reply by rozensl2** [reply]
Hi Miguel, thank you for the prompt reply! I'll be looking for another solution.
